<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using SNPknock with Genetic Data • SNPknock</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using SNPknock with Genetic Data">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SNPknock</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.8.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/SNPknock.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/genotypes.html">Using SNPknock with Genetic Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://web.stanford.edu/~msesia/">
    <span class="fa fa-user"></span>
     
    About
  </a>
</li>
<li>
  <a href="https://github.com/msesia/snpknock">
    <span class="fa fa-github"></span>
     
    GitHub
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Using SNPknock with Genetic Data</h1>
                        <h4 class="author">Matteo Sesia (<a href="mailto:msesia@stanford.edu">msesia@stanford.edu</a>)</h4>
            
            <h4 class="date">2019-05-14</h4>
      
      
      <div class="hidden name"><code>genotypes.Rmd</code></div>

    </div>

    
    
<p>This vignette illustrates the basic usage of the <code>SNPknock</code> package in combination with the phasing software <code>fastphase</code> <span class="citation">(Scheet and Stephens 2006)</span> to create knockoffs of unphased genotypes or phased haplotypes <span class="citation">(Sesia, Sabatti, and Candès 2019, <span class="citation">Sesia et al. (2019)</span>)</span>. Since <code>fastphase</code> is not available as an R package, this functionality of <code>SNPknock</code> requires the user to first obtain a copy of <code>fastphase</code>.</p>
<p>To learn more about the application of <code>SNPknock</code> to large genome-wide association studies <span class="citation">(Sesia et al. 2019)</span>, visit: <a href="https://msesia.github.io/knockoffzoom/" class="uri">https://msesia.github.io/knockoffzoom/</a>.</p>
<div id="obtaining-fastphase" class="section level2">
<h2 class="hasAnchor">
<a href="#obtaining-fastphase" class="anchor"></a>Obtaining <code>fastphase</code>
</h2>
<p><code>fastphase</code> is a phasing and imputation tool based on the hidden Markov model described in <span class="citation">(Scheet and Stephens 2006)</span>.</p>
<p>Binary executables for Linux and Mac OS are available from <a href="http://scheet.org/software.html" class="uri">http://scheet.org/software.html</a>.</p>
<p>Before continuing with this tutorial, download the <code>fastphase</code> tarball from the above link and extract the <code>fastphase</code> executable file into a convenient directory (e.g. “~/bin/”).</p>
</div>
<div id="knockoffs-for-unphased-genotypes" class="section level2">
<h2 class="hasAnchor">
<a href="#knockoffs-for-unphased-genotypes" class="anchor"></a>Knockoffs for unphased genotypes</h2>
<div id="fitting-the-hidden-markov-model-on-genotype-data" class="section level3">
<h3 class="hasAnchor">
<a href="#fitting-the-hidden-markov-model-on-genotype-data" class="anchor"></a>Fitting the hidden Markov model on genotype data</h3>
<p>A small synthetic dataset of 1454 unphased genotype SNPs from 100 individuals can be found in the package installation directory. We can load it with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(SNPknock)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">X_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"genotypes.RData"</span>, <span class="dt">package =</span> <span class="st">"SNPknock"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/load">load</a></span>(X_file)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(X)</a></code></pre></div>
<pre><code>## X
##      0      1      2 
## 130548  90112  25540</code></pre>
<p>Below, we show how to fit a hidden Markov model to this data, with the help of <code>fastphase</code>. Since <code>fastphase</code> takes as input genotype sequences in “.inp” format, we must first convert the X matrix by calling <code>writeXtoInp</code>. By default, this function will write onto a temporary file in the R temporary directory.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Convert X into the suitable fastphase input format, write it into a temporary file</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co"># and return the path to that file.</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3">Xinp_file =<span class="st"> </span><span class="kw"><a href="../reference/writeXtoInp.html">writeXtoInp</a></span>(X)</a></code></pre></div>
<p>Assuming that we have already downloaded <code>fastphase</code>, we can call it to fit the hidden Markov model to X.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">fp_path  =<span class="st"> "~/bin/fastphase"</span> <span class="co"># Path to the fastphase executable</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co"># Call fastphase and return the path to the parameter estimate files</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3">fp_outPath =<span class="st"> </span><span class="kw"><a href="../reference/runFastPhase.html">runFastPhase</a></span>(fp_path, Xinp_file, <span class="dt">K =</span> <span class="dv">12</span>, <span class="dt">numit =</span> <span class="dv">15</span>)</a></code></pre></div>
<pre><code>## ~/bin/fastphase -Pp -T1 -K12 -g -H-4 -C15 -S1 -o'/tmp/RtmpqrNhXj/file19db40ecc7eb' /tmp/RtmpqrNhXj/file19db66ab2dc2.inp</code></pre>
<p>Above, the <code>SNPknock</code> package could not find <code>fastphase</code> because we did not provide the correct path (we cannot include third-party executable files within this package). However, if you install <code>fastphase</code> separately and provide <code>SNPknock</code> with the correct path, this will work.</p>
<p>If the previous step worked for you, you can find the parameter estimates produced by <code>fastphase</code> in the following files:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">r_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(fp_outPath, <span class="st">"_rhat.txt"</span>, <span class="dt">sep=</span><span class="st">""</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">alpha_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(fp_outPath, <span class="st">"_alphahat.txt"</span>, <span class="dt">sep=</span><span class="st">""</span>)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">theta_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(fp_outPath, <span class="st">"_thetahat.txt"</span>, <span class="dt">sep=</span><span class="st">""</span>)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">char_file  =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(fp_outPath, <span class="st">"_origchars"</span>, <span class="dt">sep=</span><span class="st">""</span>)</a></code></pre></div>
<p>Otherwise, for the sake of this tutorial, you can use the example parameter files provided in the package installation directory:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">r_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"genotypes_rhat.txt"</span>, <span class="dt">package =</span> <span class="st">"SNPknock"</span>)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">alpha_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"genotypes_alphahat.txt"</span>, <span class="dt">package =</span> <span class="st">"SNPknock"</span>)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">theta_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"genotypes_thetahat.txt"</span>, <span class="dt">package =</span> <span class="st">"SNPknock"</span>)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">char_file  =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"genotypes_origchars"</span>, <span class="dt">package =</span> <span class="st">"SNPknock"</span>)</a></code></pre></div>
<p>Then, we can construct the hidden Markov model with:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">hmm =<span class="st"> </span><span class="kw"><a href="../reference/loadHMM.html">loadHMM</a></span>(r_file, alpha_file, theta_file, char_file)</a></code></pre></div>
</div>
</div>
<div id="generating-knockoff-genotypes" class="section level2">
<h2 class="hasAnchor">
<a href="#generating-knockoff-genotypes" class="anchor"></a>Generating knockoff genotypes</h2>
<p>Finally, we can use the hidden Markov model created above to generate knockoffs.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">Xk =<span class="st"> </span><span class="kw"><a href="../reference/knockoffGenotypes.html">knockoffGenotypes</a></span>(X, hmm<span class="op">$</span>r, hmm<span class="op">$</span>alpha, hmm<span class="op">$</span>theta)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(Xk)</a></code></pre></div>
<pre><code>## Xk
##      0      1      2 
## 130114  90466  25620</code></pre>
</div>
<div id="knockoffs-for-phased-haplotypes" class="section level2">
<h2 class="hasAnchor">
<a href="#knockoffs-for-phased-haplotypes" class="anchor"></a>Knockoffs for phased haplotypes</h2>
<div id="fitting-the-hidden-markov-model-on-haplotype-data" class="section level3">
<h3 class="hasAnchor">
<a href="#fitting-the-hidden-markov-model-on-haplotype-data" class="anchor"></a>Fitting the hidden Markov model on haplotype data</h3>
<p>A small synthetic dataset of 1454 phased haplotype SNPs from 100 individuals can be found in the package installation directory. We can load it with:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(SNPknock)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">H_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"haplotypes.RData"</span>, <span class="dt">package =</span> <span class="st">"SNPknock"</span>)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/load">load</a></span>(H_file)</a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(H)</a></code></pre></div>
<pre><code>## H
##      0      1 
## 351208 141192</code></pre>
<p>Below, we show how to fit a hidden Markov model to this data, with the help of <code>fastphase</code>. Since <code>fastphase</code> takes as input haplotype sequences in “.inp” format, we must first convert the H matrix by calling <code>writeXtoInp</code>. By default, this function will write onto a temporary file in the R temporary directory.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="co"># Convert X into the suitable fastphase input format, write it into a temporary file</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="co"># and return the path to that file.</span></a>
<a class="sourceLine" id="cb13-3" data-line-number="3">Hinp_file =<span class="st"> </span><span class="kw"><a href="../reference/writeXtoInp.html">writeXtoInp</a></span>(H, <span class="dt">phased =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>Assuming that we have already downloaded <code>fastphase</code>, we can call it to fit the hidden Markov model to X.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">fp_path  =<span class="st"> "~/bin/fastphase"</span> <span class="co"># Path to the fastphase executable</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="co"># Call fastphase and return the path to the parameter estimate files</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3">fp_outPath =<span class="st"> </span><span class="kw"><a href="../reference/runFastPhase.html">runFastPhase</a></span>(fp_path, Hinp_file, <span class="dt">K =</span> <span class="dv">12</span>, <span class="dt">numit =</span> <span class="dv">15</span>, <span class="dt">phased =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## ~/bin/fastphase -Pp -T1 -K12 -g -H-4 -C15 -B -S1 -o'/tmp/RtmpqrNhXj/file19db7835980b' /tmp/RtmpqrNhXj/file19db4209a588.inp</code></pre>
<p>Above, the <code>SNPknock</code> package could not find <code>fastphase</code> because we did not provide the correct path (we cannot include third-party executable files within this package). However, if you install <code>fastphase</code> separately and provide <code>SNPknock</code> with the correct path, this will work.</p>
<p>If the previous step worked for you, you can find the parameter estimates produced by <code>fastphase</code> in the following files:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">r_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(fp_outPath, <span class="st">"_rhat.txt"</span>, <span class="dt">sep=</span><span class="st">""</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">alpha_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(fp_outPath, <span class="st">"_alphahat.txt"</span>, <span class="dt">sep=</span><span class="st">""</span>)</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">theta_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(fp_outPath, <span class="st">"_thetahat.txt"</span>, <span class="dt">sep=</span><span class="st">""</span>)</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">char_file  =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(fp_outPath, <span class="st">"_origchars"</span>, <span class="dt">sep=</span><span class="st">""</span>)</a></code></pre></div>
<p>Otherwise, for the sake of this tutorial, you can use the example parameter files provided in the package installation directory:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">r_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"haplotypes_rhat.txt"</span>, <span class="dt">package =</span> <span class="st">"SNPknock"</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">alpha_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"haplotypes_alphahat.txt"</span>, <span class="dt">package =</span> <span class="st">"SNPknock"</span>)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">theta_file =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"haplotypes_thetahat.txt"</span>, <span class="dt">package =</span> <span class="st">"SNPknock"</span>)</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">char_file  =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"haplotypes_origchars"</span>, <span class="dt">package =</span> <span class="st">"SNPknock"</span>)</a></code></pre></div>
<p>Then, we can construct the hidden Markov model with:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">hmm =<span class="st"> </span><span class="kw"><a href="../reference/loadHMM.html">loadHMM</a></span>(r_file, alpha_file, theta_file, char_file)</a></code></pre></div>
</div>
<div id="generating-knockoff-haplotypes" class="section level3">
<h3 class="hasAnchor">
<a href="#generating-knockoff-haplotypes" class="anchor"></a>Generating knockoff haplotypes</h3>
<p>Finally, we can use the hidden Markov model created above to generate knockoffs.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">Hk =<span class="st"> </span><span class="kw"><a href="../reference/knockoffHaplotypes.html">knockoffHaplotypes</a></span>(H, hmm<span class="op">$</span>r, hmm<span class="op">$</span>alpha, hmm<span class="op">$</span>theta)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(Hk)</a></code></pre></div>
<pre><code>## Hk
##      0      1 
## 351155 141245</code></pre>
</div>
</div>
<div id="see-also" class="section level2">
<h2 class="hasAnchor">
<a href="#see-also" class="anchor"></a>See also</h2>
<p>If you want to see some basic usage of <code>SNPknock</code>, see the <a href="SNPknock.html">introductory vignette</a>.</p>
<p>If you want to learn about <code>SNPknock</code> for large genome-wide association studies <span class="citation">(Sesia et al. 2019)</span>, see <a href="https://msesia.github.io/knockoffzoom/" class="uri">https://msesia.github.io/knockoffzoom/</a>.</p>
<div id="refs" class="references">
<div id="ref-scheet2006">
<p>Scheet, P., and M. Stephens. 2006. “A Fast and Flexible Statistical Model for Large-Scale Population Genotype Data: Applications to Inferring Missing Genotypes and Haplotypic Phase.” <em>Am. J. Hum. Genet.</em> 78:629–44. <a href="https://doi.org/10.1086/502802" class="uri">https://doi.org/10.1086/502802</a>.</p>
</div>
<div id="ref-sesia2019multi">
<p>Sesia, M., E. Katsevich, S. Bates, E. Candès, and C. Sabatti. 2019. “Multi-Resolution Localization of Causal Variants Across the Genome.” <em>bioRxiv</em>. Cold Spring Harbor Laboratory. <a href="https://doi.org/10.1101/631390" class="uri">https://doi.org/10.1101/631390</a>.</p>
</div>
<div id="ref-sesia2019">
<p>Sesia, M., C. Sabatti, and E. J. Candès. 2019. “Gene Hunting with Hidden Markov Model Knockoffs.” <em>Biometrika</em> 106:1–18. <a href="https://doi.org/10.1093/biomet/asy033" class="uri">https://doi.org/10.1093/biomet/asy033</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#obtaining-fastphase">Obtaining <code>fastphase</code></a></li>
      <li><a href="#knockoffs-for-unphased-genotypes">Knockoffs for unphased genotypes</a></li>
      <li><a href="#generating-knockoff-genotypes">Generating knockoff genotypes</a></li>
      <li><a href="#knockoffs-for-phased-haplotypes">Knockoffs for phased haplotypes</a></li>
      <li><a href="#see-also">See also</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Matteo Sesia.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
